AK/Math.h

```diff
@@ -1015,12 +1015,16 @@ constexpr T pow(T x, T y)
         return x;
 
     // Take an integer fast path as long as the value fits within a 64-bit integer.
-    if (y >= static_cast<T>(NumericLimits<i64>::min()) && y <= static_cast<T>(NumericLimits<i64>::max())) {
+    if (y >= static_cast<T>(NumericLimits<i64>::min() + 1) && y <= static_cast<T>(NumericLimits<i64>::max())) {
```

Comment:
Hm, how are these bounds check supposed to work? Was this tested? If I try to compare values around the minimum of an `i64` I get the same results:
https://serenityos.godbolt.org/z/s61s15Yhd

Reply:
Technically integers can represent one more negative value than positive values, which would be a problem when taking the abs later,
Also technically doubles don't have the precision to represent this difference,
but I wanted to be on the safe side here
(For example long doubles can represent the difference but haven't checked those)

For if they are tested, waiting for film to test in his PR and the pure math is covered by the tests already present
For the precision gain I think there's an old issue about inaccuracies open which also suggested this change iirc

Reply:
Hm, you're trying to protect `abs(y_as_int)` from overflowing, but since the `+ 1` doesn't have an effect this fails for `float`s and `double`s. Example here:
https://serenityos.godbolt.org/z/3Yxs8Mre8

With `T = long double` things should be fine though.